Competition2017/src/competition/subsystems/BaseXCANTalonSpeedControlledSubsystem.java

```diff
@@ -0,0 +1,202 @@
+        masterMotor.reverseSensor(motorSensorInverted);
+        masterMotor.setControlMode(TalonControlMode.Speed);
+        masterMotor.setProfile(0);
+    }
+    
+    protected void initializeFollowerMotorConfiguration(boolean motorInverted) {
+        if (followerMotor == null) {
+            log.warn("initializeFollowerMotorConfiguration was called, but no followerIndex was given during creation!");
+            return;
+        }
+        followerMotor.setControlMode(TalonControlMode.Follower);
+        followerMotor.setInverted(motorInverted);
+        followerMotor.set(masterChannel);
+    }
+        
+    private void updateMotorPidValues() {
+        masterMotor.setP(pidPropertyManager.getP());
+        masterMotor.setI(pidPropertyManager.getI());
+        masterMotor.setD(pidPropertyManager.getD());
+        masterMotor.setF(pidPropertyManager.getF());
+    }
+    
+    /**
+     * Sets the output power of the system directly - no PID of any kind
+     * @param Power is set to system
+     */
+    public void setPower(double power) {
+        masterMotor.ensureTalonControlMode(TalonControlMode.PercentVbus);
+        masterMotor.set(power);
+    }
+    
+    /**
+     * Returns the last set motor power. Only works if the motor is currently in PercentVbus mode - otherwise, returns 0.
+     * @return Last Motor power
+     */
+    public double getPower() {
+        masterMotor.getOutputVoltage();
+        double inversionFactor = masterMotor.getInverted() ? -1 : 1;
+        return masterMotor.get() * inversionFactor;
+    }
+
+    /**
+     * Gives the system a new speed goal.
+     * @param systemTargetSpeed in Rotations per Second (RPS)
+     */
+    public void setTargetSpeed(double speed) {
+        masterMotor.ensureTalonControlMode(TalonControlMode.Speed);
+        
+        // Update property for dashboard
+        systemTargetSpeed.set(speed);
+        // If there have been any changes to encoder settings or PID setings, apply them.
+        updateMotorPidValues();
+        // Instruct motor about new speed goal
+        masterMotor.set(speed);
+    }
+    
+    public double getSpeed() {
+        return masterMotor.getSpeed();
+    }
+    
+    public boolean isAtSpeed() {
+        return Math.abs(getSpeed() - systemTargetSpeed.get()) <= systemSpeedThresh.get();
+    }
+    
+    public double getTargetSpeed() {
+        return systemTargetSpeed.get();
+    }
+
+    public void updatePeriodicData() {
+        masterMotor.updateTelemetryProperties();        
+        atSpeedProp.set(isAtSpeed());
+        systemCurrentSpeed.set(getSpeed());
+        systemOutputPower.set(masterMotor.getOutputVoltage() / masterMotor.getBusVoltage());
+        systemTalonError.set(masterMotor.getClosedLoopError());
+        
+        if(enablesystemLogging.get()){
+            double currentTime = Timer.getFPGATimestamp();
+            // Format: time, voltage, error, speed
+            log.info(currentTime + "," 
```

Comment:
Now that this code could be used in multiple places, it needs to specify which system it is reporting on.